pkg/ui/ccl/src/views/clusterviz/containers/map/mapLayout.tsx

```diff
@@ -29,15 +31,35 @@ interface MapLayoutProps {
   viewportSize: [number, number];
 }
 
+interface LocalityLocation {
+  locality: LocalityTree;
+  location: any;
+}
+
 interface MapLayoutState {
   zoomTransform: ZoomTransformer;
+  prevLocalityLocations: LocalityLocation[];
 }
 
 export class MapLayout extends React.Component<MapLayoutProps, MapLayoutState> {
   gEl: any;
   zoom: d3.behavior.Zoom<any>;
   maxLatitude = 80;
 
+  localityLocations = createSelector(
+    (props: MapLayoutProps) => props.localityTree,
+    (props: MapLayoutProps) => props.locationTree,
+    (localityTree, locationTree) => {
+      return _.map(getChildLocalities(localityTree), locality => {
+        const location = findOrCalculateLocation(locationTree, locality);
+        return {
+          locality,
+          location,
+        };
+      });
+    },
+  );
```

Comment:
nice. Eventually let's move this up to one of the selectors files so we can use it from the locality tree debug page and elsewhere.